src/microbe_stage/CellHexPhotoBuilder.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using Godot;
+
+public class CellHexPhotoBuilder : Spatial, IPhotographable
+{
+    private float radius;
+    private bool radiusDirty;
+    private MicrobeSpecies? species;
+
+    public string SceneToPhotographPath => "res://src/microbe_stage/CellHexPhotoBuilder.tscn";
+
+    public float Radius
+    {
+        get
+        {
+            if (radiusDirty)
+                CalculateRadius();
+
+            return radius;
+        }
+    }
+
+    public MicrobeSpecies? Species
+    {
+        get => species;
+        set
+        {
+            species = value;
+            radiusDirty = true;
+        }
+    }
+
+    public void ApplySceneParameters(Spatial instancedScene)
+    {
+        var builder = (CellHexPhotoBuilder)instancedScene;
+        builder.Species = Species;
+        builder.BuildHexStruct();
+    }
+
+    public float CalculatePhotographDistance(Spatial instancedScene)
+    {
+        return PhotoStudio.CameraDistanceFromRadiusOfObject(((CellHexPhotoBuilder)instancedScene).Radius *
+            Constants.PHOTO_STUDIO_CELL_RADIUS_MULTIPLIER);
+    }
+
+    private void BuildHexStruct()
+    {
+        var hexScene = GD.Load<PackedScene>("res://src/microbe_stage/editor/EditorHex.tscn");
+        var hexMaterial = GD.Load<Material>("res://src/microbe_stage/editor/ValidHex.material");
+        var modelScene = GD.Load<PackedScene>("res://src/general/SceneDisplayer.tscn");
+
+        this.QueueFreeChildren();
+
+        if (Species == null)
+            throw new InvalidOperationException("Species is ont initialized");
+
+        var organelleLayout = Species.Organelles.Organelles;
```

Comment:
The duplication reduction is still not perfect so could you at least add a TODO comment here pointing out the duplication with the other place? And also please add a comment in that editor about the duplication that if the code there is changed CellHexesPhotoBuilder also needs those changes.